### PR TITLE
#187854921: Add Seller and Admin Dashboard Features

### PR DESCRIPTION
**Pull request by tchamianest**
#### What does this PR do?
Dashboard
#### Description of Task to be completed?
Two different dashboards based on the type of user who are logged in and all work smoothly with functionalities.
#### How should this be manually tested?
- localhost : npm run dev
-  deployed version preview 
#### What are the relevant pivotal tracker/Trello stories?
https://www.pivotaltracker.com/n/projects/2698211/stories/187854921
#### Screenshots (if appropriate)
image

### CODE REVIEW

**Review by kayigmb — APPROVED** [review]

File: src/app/(profile)/profile-edit/page.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-"use client";
```
Comment (Eli250) [inline src/app/(profile)/profile-edit/page.tsx:1]: This endpoint `src/app/(profile)/profile-edit/...` does not not follow our naming conventions. It has to be something close to `.../profile/[profileId]`.

File: src/app/(profile)/profile-edit/page.tsx [hunk 2]
```diff
@@ -97,121 +107,251 @@ const UserProfileForm: React.FC = () => {
...
       if (updateUserProfile.fulfilled.match(response)) {
         // Update only the User value in localStorage
-        const currentProfile = JSON.parse(localStorage.getItem('profile') || '{}');
+        const currentProfile = JSON.parse(
+          localStorage.getItem('profile') || '{}',
+        );
         if (response.payload && response.payload.User) {
           currentProfile.User = response.payload.User;
-          console.log("currentProfile", currentProfile);
+          console.log('currentProfile', currentProfile);
```
Comment (Eli250) [inline src/app/(profile)/profile-edit/page.tsx:115]: Remove these consoles.

File: src/app/(profile)/profile-edit/page.tsx [hunk 3]
```diff
@@ -97,121 +107,251 @@ const UserProfileForm: React.FC = () => {
...
+                <p className="text-sm text-gray-500 mb-4 text-center md:text-left">
+                  {user?.User?.Role.name || 'buyer'}
+                </p>
               </div>
 
-              <ul className='hidden md:block text-sm space-y-2'>
-                <li className='flex items-center text-gray-600'>
-                  <svg className='w-4 h-4 mr-2' fill='none' stroke='currentColor' viewBox='0 0 24 24' xmlns='http://www.w3.org/2000/svg'><path strokeLinecap='round' strokeLinejoin='round' strokeWidth={2} d='M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z' /></svg>
-                  {watchedValues.birthDate || "YYYY-MM-DD"}
+              <ul className="hidden md:block text-sm space-y-2">
```
Comment (Eli250) [inline src/app/(profile)/profile-edit/page.tsx:200]: This file is too long. You are probably not using reusable components. Refactor this component by dividing it into smaller components that perform specific tasks.

File: src/app/dashboard/addproduct/page.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+'use client';
```
Comment (Eli250) [inline src/app/dashboard/addproduct/page.tsx:1]: We have never accepted this endpoint (`src/app/dashboard/addproduct/page.tsx`). You need to revise our naming conventions.

File: src/app/dashboard/home/page.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
+'use client';
```
Comment (Eli250) [inline src/app/dashboard/home/page.tsx:1]: Why can't you just use `/dashboard`. The home is not needed.

File: src/app/dashboard/product_/[id]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
+'use client';
+import React, { useEffect, useState } from 'react';
```
Comment (Eli250) [inline src/app/dashboard/product_/[id]/page.tsx:2]: This page endpoint is wrong, and I have mentioned this several times.

File: src/app/dashboard/product_/[id]/page.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  const [user, setUser] = useState<any>();
+  const [product, setProduct] = useState();
+  const router = useRouter();
+  const { id } = useParams();
+
+  const productId: any = id;
+
+  useEffect(() => {
+    const categ = async () => {
+      const responsecat: any = await request.get(`/products/${productId}`);
```
Comment (Eli250) [inline src/app/dashboard/product_/[id]/page.tsx:18]: Do not use `any`

File: src/components/2faVerification.tsx [hunk 8]
```diff
@@ -32,7 +31,7 @@ const OtpVerify: React.FC<OtpVerifyInterface> = ({ isOpen }) => {
 
   useEffect(() => {
     if (isAuthenticated) {
-      router.push('/seller/dashboard/products');
+      router.push('/dashboard/home');
```
Comment (Eli250) [inline src/components/2faVerification.tsx:34]: `/dashboard/home` endpoint does not make sense.

File: src/components/EditProfile.tsx [hunk 9]
```diff
@@ -0,0 +1,378 @@
...
+                </div>
+              </form>
+            </div>
+          </div>
+        </div>
+      </main>
+    </div>
+  );
+};
+
```
Comment (Eli250) [inline src/components/EditProfile.tsx:377]: This component is too long. You have to rafactor it.

**Review by Eli250 — CHANGES_REQUESTED** [review]
Resolve all feedback.

**Pull request merged**